mp/src/game/client/momentum/ui/chat/ChatLine.cpp

```diff
@@ -0,0 +1,336 @@
+#include "cbase.h"
+
+#include "ChatLine.h"
+
+
+#include "ChatHistory.h"
+#include "ChatPanel.h"
+
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+static ConVar hud_saytext_time("hud_saytext_time", "12", 0);
+
+ChatLine::ChatLine(ChatPanel *parent) : RichText(parent, "ChatLine")
+{
+    m_pChat = parent;
+    m_nLineCounter = 1;
+
+    m_hFont = m_hFontMarlett = 0;
+    m_flExpireTime = 0.0f;
+    m_flStartTime = 0.0f;
+    m_iNameLength = 0;
+    m_text = NULL;
+
+    SetPaintBackgroundEnabled(true);
+
+    SetVerticalScrollbar(false);
+}
+
+ChatLine::~ChatLine()
+{
+    if (m_text)
+    {
+        delete[] m_text;
+        m_text = NULL;
+    }
+}
+
+void ChatLine::ApplySchemeSettings(IScheme *pScheme)
+{
+    BaseClass::ApplySchemeSettings(pScheme);
+
+    m_hFont = pScheme->GetFont("Default");
+
+    SetBgColor(Color(0, 0, 0, 100));
+
+    m_hFontMarlett = pScheme->GetFont("Marlett");
```

Comment:
Why Marlett? I don't see `m_hFontMarlett` being used anywhere anyways.

Reply:
Good catch, it's not used. Removed it.